src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1874,7 +1874,7 @@ private void EnsureNameofExpressionSymbols(BoundMethodGroup methodGroup, Binding
         /// <summary>
         /// Returns true if syntax form is OK (so no errors were reported)
         /// </summary>
-        private bool CheckSyntaxForNameofArgument(ExpressionSyntax argument, out string name, BindingDiagnosticBag diagnostics, bool top = true)
+        private bool CheckSyntaxForNameofArgument(ExpressionSyntax argument, out string name, BindingDiagnosticBag diagnostics, bool boundArgumentHasAnyErrors, bool top = true)
```

Comment:
>bool boundArgumentHasAnyErrors [](start = 128, length = 30)

Instead of adding this parameter we should consider passing BindingDiagnosticBag.Discarded when we are trying to suppress cascading diagnostics.

Reply:
@AlekseyTs That simplifies it **alot**! 🎉